File: app/src/main/java/rssdemo/andras/hu/rssdemo/ui/subscriptions/editor/SubscriptionEditorView.java
package rssdemo.andras.hu.rssdemo.ui.subscriptions.editor;


import android.support.annotation.StringRes;

import rssdemo.andras.hu.rssdemo.data.Subscription;

public interface SubscriptionEditorView {

    void raiseErrorOnName(@StringRes int messageResId);

    void raiseErrorOnUrl(@StringRes int messageResId);

    void closeEditor();

    Subscription getSubscrition();
}

File: app/src/main/java/rssdemo/andras/hu/rssdemo/ui/subscriptions/editor/SubscriptionEditorView.kt
package rssdemo.andras.hu.rssdemo.ui.subscriptions.editor


import android.support.annotation.StringRes

import rssdemo.andras.hu.rssdemo.data.Subscription

internal interface SubscriptionEditorView {

    fun raiseErrorOnName(@StringRes messageResId: Int)

    fun raiseErrorOnUrl(@StringRes messageResId: Int)

    fun closeEditor()

    fun getSubscription(): Subscription
}
